app/Models/WorkshopBalance.php

```diff
@@ -24,4 +27,30 @@ public function semester()
     {
         return $this->belongsTo('App\Models\Semester');
     }
+
+    /**
+     * Generates all the workshops' allocated balance in the current semester.
+     * For all active members in a workshop who payed kkt: 
+     *      kkt * (isResident ? 0.6 : 0.45) / member's workshops' count
+    */
+    public static function generateBalances()
+    {
+        foreach(Workshop::all() as $workshop)
+        {
+            $balance = 0;
+            foreach ($workshop->users as $member) {
+                if ($member->isActive() && !$member->hasToPayKKTNetreg())
+                {
+                    $balance += config('custom.kkt') *
+                        ($member->isResident() ? 0.6 : 0.45 ) /
```

Comment:
These should come from env --> config

Reply:
What do you mean?

Reply:
The proportion should come from env, with a default value in the config.

Reply:
It is coming from env

Reply:
Just the values were missing from .env.example, I added them now.

Reply:
I am talking about 0.6 and 0.45 hardcoded here.

Reply:
Oh, ok 👍